WordPress/gutenberg
# Fix tsconfig for test/ directory

## What?
Currently, the files inside [`test/`](https://github.com/WordPress/gutenberg/tree/5c7f039bf6804449ec09723f37179d76355c14d7/test) directory are not checked for TS errors by our static type checking which means if any TS error is introduced in those files, we won't know about it.
This PR:
- Adds `@types/node` as a dependency to allow its inclusion in `tsconfig.json` files in `test/e2e` and `test/performance` directories
- Updates and cleans up the `tsconfig.json` files for e2e and performance test directories to add `rootDir`, which was not done in https://github.com/WordPress/gutenberg/pull/68326#issuecomment-2563518504
- Adds `test/e2e` and `test/performance` to root `tsconfig.json` references to include them for typechecking

## Why?

To ensure that the test files are type checked

## How?


## Testing Instructions
- On trunk make any type error in any TS file in `test/e2e/specs`
- Run `pm run build:package-types`
- Notice that the error won't be reported
- Now on this PR run the command again
- Confirm that the error is reported

### Testing Instructions for Keyboard


## Screenshots or screencast 



|Before|After|
|-|-|
|||
